TEST_CHECKLIST_GUIDELINES.md

```diff
@@ -0,0 +1,56 @@
+Testing Scripts should
+- Be easy to follow
+- Be grouped in a logical way (either page or overarching component, like "Login" or "Filters" or "Saved search") as much as possible
+- Reference other tests or groups of tests (ie: when we are done testing saved searches, we make sure that collections are also OK since we might have impacted them) 
+- Assume a basic comprehension of the system, to not be overly verbose. 
+
+    ie: we can assume that the user knows how to login if that's a feature we have had for a long time, OR at least link to the login test checklist in the "Login" instruction if we need to be more specific 
+
+- Explicitly test for non happy paths
+
+ie:  If you are desgining a login test checklist, it should include validation errors
+
+- Reflect problems we had in the past (while doing dev or testing, etc), to keep adding to our test suites as we move forward with the platform development.  
+    
+    Example: if you faced a bug like "the filters don't close when I click the x button" make sure you create a checklist for that. 
```

Comment:
So this sounds like a mix of two elements, one is a side effect "weirdness", the second is the failure of a very small behaviour - in this case the x - I'm not sure I follow how either of those would become a checklist. We can require that a checklist clicks every single button but it seems more like the failure of a button like that is a systemic issue. All x buttons should be governed by the same code to try to reduce the possibility of this happening. If they are and something that changed, unrelated to the code that controls the x, caused the bug then the only way I imagine we can control for that is to require that ALL buttons within an interface that changed be clicked - that seems like the reason headless browser style testing existed.

May be we need to work on defining a way to move things like - "every button should pass clicking" - into basic page tests. We had those kind of tests originally but we were over using them and they became very heavy. We might have been better off just using them for extremely basic behaviours.

If the issue was caused by some kind of weird side effect, I would worry that by adding every odd case to the checklists we would create a lot of wasted time on elements that were once off bugs that should never repeat. Instead it might be better if we can stick to structure and classing the issues. I'd be happy to chat about this in order to fully explain my thinking. But in short I think, "This is the set of behaviours expected in this view as a checklist" is better than "Along with all the other tests on this checklist also make sure you try to reproduce this weird bug that happened once a long time ago that is unrelated to the interfaces behaviour". 

It seems like the latter would be better captured through super simple headless browser tests.